fcli-core/fcli-tool/src/main/java/com/fortify/cli/tool/definitions/helper/ToolDefinitionsHelper.java

```diff
@@ -34,104 +38,264 @@
-        String yamlFileName = toolName + ".yaml";
-        try ( InputStream is = getToolDefinitionsInputStream(); ZipInputStream zis = new ZipInputStream(is) ) {
-            ZipEntry entry;
-            while ( (entry = zis.getNextEntry())!=null ) {
-                if ( yamlFileName.equals(entry.getName()) ) {
-                    return yamlObjectMapper.readValue(zis, ToolDefinitionRootDescriptor.class);
-                }
-            }
-            throw new FcliSimpleException("No tool definitions found for "+toolName);
-        } catch (IOException e) {
-            throw new FcliSimpleException("Error loading tool definitions", e);
-        }
-    }
-    
-    private static final InputStream getToolDefinitionsInputStream() throws IOException {
-        return Files.exists(DEFINITIONS_STATE_ZIP) 
-                ? Files.newInputStream(DEFINITIONS_STATE_ZIP) 
-                : FileUtils.getResourceInputStream(DEFINITIONS_INTERNAL_ZIP);
-    }
-    
-    private static final void addZipOutputDescriptor(List<ToolDefinitionsOutputDescriptor> result) {
-        var stateDescriptor = FcliDataHelper.readFile(DESCRIPTOR_PATH, ToolDefinitionsStateDescriptor.class, false);
-        if ( stateDescriptor!=null ) {
-            result.add(new ToolDefinitionsOutputDescriptor(ZIP_FILE_NAME, stateDescriptor));
-        } else {
-            result.add(new ToolDefinitionsOutputDescriptor(ZIP_FILE_NAME, "INTERNAL", FcliBuildProperties.INSTANCE.getFcliBuildDate()));
-        }
-    }
-    
-    private static final void addYamlOutputDescriptors(List<ToolDefinitionsOutputDescriptor> result) {
-        try ( InputStream is = getToolDefinitionsInputStream(); ZipInputStream zis = new ZipInputStream(is) ) {
-            ZipEntry entry;
-            while ( (entry = zis.getNextEntry())!=null ) {
-                var name = Path.of(entry.getName()).getFileName().toString(); // Should already be just a file name, but just in case
-                var source = ZIP_FILE_NAME;
-                var lastModified = new Date(entry.getLastModifiedTime().toMillis());
-                result.add(new ToolDefinitionsOutputDescriptor(name, source, lastModified));
-            }
-        } catch (IOException e) {
-            throw new FcliSimpleException("Error loading tool definitions", e);
-        }
-    }
-    
-}
+	private static final String ZIP_FILE_NAME = "tool-definitions.yaml.zip";
+	public static final Path DEFINITIONS_STATE_DIR = FcliDataHelper.getFcliStatePath().resolve("tool");
+	public static final Path DEFINITIONS_STATE_ZIP = DEFINITIONS_STATE_DIR.resolve(ZIP_FILE_NAME);
+	private static final String DEFINITIONS_INTERNAL_ZIP = "com/fortify/cli/tool/config/" + ZIP_FILE_NAME;
+	private static final Path DESCRIPTOR_PATH = ToolDefinitionsHelper.DEFINITIONS_STATE_DIR.resolve("state.json");
+	private static final ObjectMapper yamlObjectMapper = new ObjectMapper(new YAMLFactory());
+	private static String toolDefinitionCustomFilePath;
```

Comment:
Using a static mutable field like toolDefinitionCustomFilePath is generally considered bad practice. It introduces shared state across all usages of the class, which can lead to race conditions, unexpected side effects, and makes the code harder to test and maintain. Prefer passing such values as method parameters or encapsulating them in an instance field within a properly scoped object.